### PR TITLE
fix(TextArea): remove invalid `onFocus` and `onBlur` props

### LINKED ISSUE #10192: Bug - TextArea - The `onFocus` and `onBlur` props are overwritten with an invalid type signature
**Describe the problem**
The `onFocus` and `onBlur` props [are overwritten](https://github.com/patternfly/patternfly-react/blob/bb7af9ae8ad9b31892f14878c9bb30599bd88bcd/packages/react-core/src/components/TextArea/TextArea.tsx#L39-L42) in the `TextArea` component. However, the `event` parameter for these callbacks has been made optional in the overwritten type. This causes issues with libraries that expect this type signature to math the ones from a native React `<textarea>` element.

**How do you reproduce the problem?**
Set up a form using [React Hook Form](https://react-hook-form.com/) with a PatternFly `TextArea` component in a TypeScript environment, for example:

```tsx
import { TextArea } from "@patternfly/react-core";
import { useForm } from "react-hook-form";

export default function App() {
  const { register handleSubmit } = useForm();

  return (
    <TextArea {...register("example")} />
  )
}
```

**Expected behavior**
No compilation errors occur.

**Actual behaviour**
The compilation fails on a type error on the mismatched signatures of `onFocus` and `onBlur`.

**Is this issue blocking you?**
Yes, this is preventing us from upgrading to PatternFly 5.

**Screenshots**
A screenshot showing the aforementioned type error.

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #10192

### CODE REVIEW

**Comment by nicolethoen** [comment]
@tlabaj do you think this is a breaking change? Or is it okay like this since it's bug?

@jonkoops I wonder if the type could alternatively be updated to a union of the correct type rather than removing it to minimize impact to consumers?

**Review by kmcfaul — COMMENTED** [review]
I'm more of a fan of fixing the signatures rather than removing them. I also don't believe this truly counts as a breaking change because its a bug, but I'll defer to @tlabaj

**Review by kmcfaul — APPROVED** [review]

**Comment by kmcfaul** [comment]
Removing for v5 to unblock, adding back with proper signature for v6 #10215

**Review by tlabaj — APPROVED** [review]

**Pull request merged**
